src/data/list/zip.lean

```diff
@@ -87,6 +87,28 @@ theorem zip_map_right (f : β → γ) (l₁ : list α) (l₂ : list β) :
    zip l₁ (l₂.map f) = (zip l₁ l₂).map (prod.map id f) :=
 by rw [← zip_map, map_id]
 
+lemma zip_with_map_left
+  (f : α → β → γ) (g : δ → α) (l : list δ) (l' : list β) :
+  zip_with f (l.map g) l' = zip_with (f ∘ g) l l' :=
```

Comment:
I would add a lemma

```lean
zip_with_map : zip_with f (as.map g) (bs.map h) = zip_with (λ a b, f (g a) (h b)) as bs
```

The `_left` and `_right` lemmas are then consequences of `zip_with_map` with, respectively, `f = id` and `g = id`.

Reply:
It is now done!

Reply:
If you see shorter "term-mode" proofs for the left/right lemmas, I'd be glad to use them